### PR TITLE
zingolib updatedcurrentprice command with NaN value fixed

### LINKED ISSUE #264: Gemini is down - Reflect this in the UI if the user fetch the price
In a few places in the App the user can fetch the price from Gemini, including the `server info screen`.

It would be nice if the user have a `toast` or message with info about what is happening in Gemini and Why I get 0 as a USD price.
And if the user go to `Server Info Screen` would see the reason why the price is 0. Here the explanation could be with more details...

### PR DESCRIPTION
**Pull request by juanky201271**
When zingo-mobile runs the zingolib command: `updatedcurrentprice` and the result (whatever) is not a Number -> returns 0, otherwise return the value parsed to float.

TODO: https://github.com/zingolabs/zingo-mobile/issues/264

### CODE REVIEW

**Review by fluidvanadium — COMMENTED** [review]
i dont agree that the fallback value should be 0. what about -1?

**Comment by juanky201271** [comment]
OK, I like -1 as a value when `gemini` is down. I'll keep working on this PR & I'll resolve this issue as well... https://github.com/zingolabs/zingo-mobile/issues/264, thanks @fluidvanadium

**Comment by juanky201271** [comment]
> i dont agree that the fallback value should be 0. what about -1?

I did the changes to use -1 value when the price fetcher fails. Ready for review, @fluidvanadium .

**Comment by fluidvanadium** [comment]
those values can be different, so lets make them different! can we use -1, -2, -3, etc?

**Comment by juanky201271** [comment]
> those values can be different, so lets make them different! can we use -1, -2, -3, etc?

I don't understand what you talking about. Can you explain to me for what we can use -2, -3, etc... I don't get it. (I only understand -1 as a `error` value from zingolib/gemini, but the others negative values...)

File: app/rpc/RPC.ts [hunk 1]
```diff
@@ -97,24 +97,29 @@ export default class RPC {
...
+    // > 0 - real value
     const resultStr: string = await RPCModule.execute('updatecurrentprice', '');
     //console.log(resultStr);
 
     if (resultStr) {
-      if (resultStr.toLowerCase().startsWith('error')) {
+      if (resultStr.toLowerCase().startsWith('error') || isNaN(parseFloat(resultStr))) {
         //console.log(`Error fetching price ${resultStr}`);
-        return 0;
+        return -1;
```
Comment (fluidvanadium) [inline app/rpc/RPC.ts:116]: to me, it is more helpful if this value

File: app/rpc/RPC.ts [hunk 2]
```diff
@@ -97,24 +97,29 @@ export default class RPC {
...
       } else {
         return parseFloat(resultStr);
       }
+    } else {
+      //console.log(`Error fetching price ${resultStr}`);
+      return -1;
```
Comment (fluidvanadium) [inline app/rpc/RPC.ts]: differs from this value, but it probably doesnt matter

**Review by fluidvanadium — DISMISSED** [review]

File: app/rpc/RPC.ts (hunk 2 above)
Comment (juanky201271) [reply app/rpc/RPC.ts]: Thanks for the explanation, now, I got it... I'm going to change it with two different texts in the info screen.

**Comment by juanky201271** [comment]
I pushed a commit with the change you ask me and I need you approval again, sorry & thanks. @fluidvanadium

**Comment by zancas** [comment]
So this is waiting for @fluidvanadium 's review?

**Review by fluidvanadium — APPROVED** [review]
yes! super cool! i love it

**Pull request merged**